### PR TITLE
feat(badge): add proposal

### PR DESCRIPTION
**Pull request by assuncaocharles**
This PR is following #223 adding the initial proposal for `Badge`

### CODE REVIEW

File: research/src/pages/badge.proposal.mdx [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+## &lt;badge&gt; Properties <a href="#badge-properties" id="badge-properties"></a>
+
+| Attribute Name | Type      | Default Value | Description                                                 |     |
+| -------------- | --------- | ------------- | ----------------------------------------------------------- | --- |
+| `value`        | `string`  | `""`          | Text value to be displayed by badge                         |     |
+| `circular`     | `boolean` | `true`        | If the badge shape is circular                              |     |
+| `color`        | `string`  | `red`         | Badge background color                                      |     |
+| `empty`        | `boolean` | `false`       | Badge can be rendered as a dot                              |     |
+| `ribbon`       | `string`  | `right`       | Badge can appear as a ribbon attaching itself to an element |     |
```
Comment (una) [inline research/src/pages/badge.proposal.mdx:38]: Can you explain the need for `ribbon`? Is this presentational or functional?

File: research/src/pages/badge.proposal.mdx [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+- [Lightning](https://www.lightningdesignsystem.com/components/badges/)
+
+## &lt;badge&gt; Properties <a href="#badge-properties" id="badge-properties"></a>
+
+| Attribute Name | Type      | Default Value | Description                                                 |     |
+| -------------- | --------- | ------------- | ----------------------------------------------------------- | --- |
+| `value`        | `string`  | `""`          | Text value to be displayed by badge                         |     |
+| `circular`     | `boolean` | `true`        | If the badge shape is circular                              |     |
+| `color`        | `string`  | `red`         | Badge background color                                      |     |
+| `empty`        | `boolean` | `false`       | Badge can be rendered as a dot                              |     |
```
Comment (una) [inline research/src/pages/badge.proposal.mdx:37]: Why would we need an `empty` state? could we simple have `value`, and hook into if `value` is empty?

**Comment by una** [comment]
Thanks for this PR :)

**Review by una — CHANGES_REQUESTED** [review]
Left some questions to discuss, thank you for opening this PR

File: research/src/pages/badge.proposal.mdx [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+- [FAST DNA](https://explore.fast.design/components/fast-badge)
+- [Atlas Kit](https://atlassian.design/components/badge/examples)
+- [Lightning](https://www.lightningdesignsystem.com/components/badges/)
+
+## &lt;badge&gt; Properties <a href="#badge-properties" id="badge-properties"></a>
+
+| Attribute Name | Type      | Default Value | Description                                                 |     |
+| -------------- | --------- | ------------- | ----------------------------------------------------------- | --- |
+| `value`        | `string`  | `""`          | Text value to be displayed by badge                         |     |
+| `circular`     | `boolean` | `true`        | If the badge shape is circular                              |     |
```
Comment (una) [inline research/src/pages/badge.proposal.mdx:35]: circular also feels like the type of thing that would be adjusted with CSS< not an attribute

File: research/src/pages/badge.proposal.mdx [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+| `color`        | `string`  | `red`         | Badge background color                                      |     |
+| `empty`        | `boolean` | `false`       | Badge can be rendered as a dot                              |     |
+| `ribbon`       | `string`  | `right`       | Badge can appear as a ribbon attaching itself to an element |     |
+| `size`         | `string`  | `small`       | Badge can have different sizes                              |     |
```
Comment (una) [inline research/src/pages/badge.proposal.mdx:39]: Another question RE: CSS being the API for sizing and other styling rather than attributes

**Review by una — CHANGES_REQUESTED** [review]

File: research/src/pages/badge.proposal.mdx (hunk 1 above)
Comment (assuncaocharles) [reply research/src/pages/badge.proposal.mdx:38]: Have a look into `AntD` ribbon badge example: https://ant.design/components/badge/#components-badge-demo-ribbbon

File: research/src/pages/badge.proposal.mdx (hunk 2 above)
Comment (assuncaocharles) [reply research/src/pages/badge.proposal.mdx:37]: It can be a good strategy

Comment (assuncaocharles) [inline research/src/pages/badge.proposal.mdx:37]:
```suggestion
```

File: research/src/pages/badge.proposal.mdx (hunk 3 above)
Comment (assuncaocharles) [reply research/src/pages/badge.proposal.mdx:35]: I feel like it's common practice within most of the libs to have this easy way to set it.

File: research/src/pages/badge.proposal.mdx (hunk 4 above)
Comment (assuncaocharles) [reply research/src/pages/badge.proposal.mdx:38]: Having pre default values are good in several moments if it's for easier use or for css cache...

**Comment by andrico1234** [comment]
I'm closing this due to inactivity, but if anyone would like to champion this MR, feel free to reopen 😄

**Pull request closed without merging**